lua/nvim-treesitter/query.lua

```diff
@@ -79,36 +69,48 @@ local function filtered_runtime_queries(lang, query_name)
   return filter_files(api.nvim_get_runtime_file(string.format('queries/%s/%s.scm', lang, query_name), true) or {})
 end
 
-local function runtime_query_exists(lang, query_name)
-  local files = api.nvim_get_runtime_file(string.format('queries/%s/%s.scm', lang, query_name), false)
-  return files and #files > 0
-end
+local function get_query_files(lang, query_name, is_included)
+  local lang_files = filtered_runtime_queries(lang, query_name)
+  local query_files = lang_files
 
-local function get_query_files(lang, query_name)
-  local query_files = {}
+  if #query_files == 0 then return {} end
 
-  local lang_files = filtered_runtime_queries(lang, query_name)
-  vim.list_extend(query_files, lang_files)
+  local base_langs = {}
 
-  for _, base_lang in ipairs(M.base_language_map[lang] or {}) do
-    local base_files = filtered_runtime_queries(base_lang, query_name)
+  -- Now get the base languages by looking at the first line of every file
+  for _, file in ipairs(query_files) do
+    local modeline = vim.fn.readfile(file, "", 1)
+
+    if #modeline == 1 then
+      local langlist = modeline[1]:match("^;%s*inherits%s*:%s*([a-z_,()]+)$")
```

Comment:
Allow multiple `;`? Would be nice if @steelsojka could highlight the magic of this comment in the query parser.

Reply:
Yeah that would be nice

Reply:
Addressed in latest push.

Reply:
you must change or remove the SHA in the lockfile (until we have 0.17.1 nvim we can make a general update of the lockfile)